routing/validation_barrier.go

```diff
@@ -44,7 +44,7 @@ type ValidationBarrier struct {
 
 // NewValidationBarrier creates a new instance of a validation barrier given
 // the total number of active requests, and a quit channel which will be used
-// to know when to kill an pending, but unfilled jobs.
+// to know when to kill a pending, but unfilled jobs.
```

Comment:
The "a" should be dropped completely as its referring to "jobs", which is plural.